csharp/autobuilder/Semmle.Autobuild.CSharp/DotNetRule.cs

```diff
@@ -94,6 +94,7 @@ public static BuildScript WithDotNet(IAutobuilder<AutobuildOptionsShared> builde
                     env = new Dictionary<string, string>{
                             { "DOTNET_MULTILEVEL_LOOKUP", "false" }, // prevent look up of other .NET Core SDKs
                             { "DOTNET_SKIP_FIRST_TIME_EXPERIENCE", "true" },
+                            { "MSBUILDDISABLENODEREUSE", "1" },
```

Comment:
This looks good to me, but I was wondering whether we want to include the variable in 

https://github.com/github/codeql/blob/e7676a00d21fd2a326b391af60d258ebae3a0dc0/csharp/codeql-extractor.yml#L7-L8

Reply:
I don't know what `extra_env_vars` is. Would that also apply to traced extraction? The current change only has an impact on autobuild and things that are started by autobuild (such as the roslyn based buildless)

Reply:
Yes that also applies to traced extraction. Not sure if that is desirable or not, but it might prevent similar issues when running explicit build commands.

Reply:
Let's keep it as it is, while keeping in mind that we have other options to add the environment variable.